### PR TITLE
[en] Add style rule OCCASION_TRANSITIVE_VERB_ARCHAIC

### PR DESCRIPTION
**Pull request by hftf**
Add a style rule to warn against using the transitive verb "to occasion," which is [archaic](https://books.google.com/ngrams/graph?content=caused+by%2Coccasioned+by%2Cinduced+by%2Cbrought+about+by%2Cinfluenced+by%2Cprovoked+by&amp;year_start=1800&amp;year_end=2008&amp;corpus=15&amp;smoothing=3&amp;share=&amp;direct_url=t1%3B%2Ccaused%20by%3B%2Cc0%3B.t1%3B%2Coccasioned%20by%3B%2Cc0%3B.t1%3B%2Cinduced%20by%3B%2Cc0%3B.t1%3B%2Cbrought%20about%20by%3B%2Cc0%3B.t1%3B%2Cinfluenced%20by%3B%2Cc0%3B.t1%3B%2Cprovoked%20by%3B%2Cc0#t1%3B%2Ccaused%20by%3B%2Cc0%3B.t1%3B%2Coccasioned%20by%3B%2Cc0%3B.t1%3B%2Cinduced%20by%3B%2Cc0%3B.t1%3B%2Cbrought%20about%20by%3B%2Cc0%3B.t1%3B%2Cinfluenced%20by%3B%2Cc0%3B.t1%3B%2Cprovoked%20by%3B%2Cc0). Some examples:

> The concert **occasioned** a riot.
> It was **occasioned** by it.
> It can be used to **occasion** it.

This is a draft pull request, so it should probably be thoroughly tested and reviewed before marked as ready to merge.

### CODE REVIEW

File: languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml [hunk 1]
```diff
@@ -83369,6 +83369,48 @@ The accident victim died from her injuries.
...
+				--><suggestion><match no="1" postag="VB.*">cause</match></suggestion>, <!--
+				--><suggestion><match no="1" postag="VB.*">induce</match></suggestion>, or <!--
+				--><suggestion><match no="1" postag="VB.*">bring</match> about</suggestion>
+			</message>
+			<url>https://tinyurl.com/yyxj28z8</url>
+			<example correction="caused|induced|brought about">The concert <marker>occasioned</marker> a riot.</example>
+			<example correction="caused|induced|brought about">It was <marker>occasioned</marker> by it.</example>
+			<!-- Not sure yet how to code a suggestion like 'bring [direct object] about'
+			(suggestion should give 'bring it about', not just 'bring about').
+			Would I need to extend the pattern <marker> around the entire direct object? -->
```
Comment (udomai) [inline languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:83396]: Yes, you would indeed have to put both the predicate and the direct object inside the pattern. My impression is that English avoids excessively long "brackets" and only requires _about_ to be put to the right of the object if it is one (or two?) words long. Is that so?

_It was used to occasion it._ --> It was used to bring it about.
_It was used to occasion those changes._ --> It was used to bring those changes about / bring about those changes.
_It was used to occasion those long-needed changes in our society._ --> It was used to bring about those long-needed changes in our society (not: It was used to **bring** those long-needed changes in our society **about**)

File: languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml [hunk 2]
```diff
@@ -83369,6 +83369,48 @@ The accident victim died from her injuries.
...
+			</antipattern>
+			<pattern>
+				<marker><token postag="VB.*" postag_regexp="yes" inflected="yes">occasion</token></marker>
+			</pattern>
+			<message>The transitive verb "<match no="1" />" is an archaic word choice. Consider a more common synonym, such as <!--
+				--><suggestion><match no="1" postag="VB.*">cause</match></suggestion>, <!--
+				--><suggestion><match no="1" postag="VB.*">induce</match></suggestion>, or <!--
+				--><suggestion><match no="1" postag="VB.*">bring</match> about</suggestion>
+			</message>
+			<url>https://tinyurl.com/yyxj28z8</url>
```
Comment (danielnaber) [inline languagetool-language-modules/en/src/main/resources/org/languagetool/rules/en/grammar.xml:83391]: This is interesting, but we should use the `url` to point to explanations, not to pure data. Also, it's okay to use long URLs, no need for tinyurl.

**Comment by MikeUnwalla** [comment]
The transitive verb 'occasion' is formal, not archaic. It might be rare, but it is not wrong:

https://www.merriam-webster.com/dictionary/occasion
https://www.lexico.com/definition/occasion

**Comment by udomai** [comment]
> It might be rare, but it is not wrong:

That is true. Indeed, this is a draft for a style suggestion (it will recommend against the usage of this word except for very formal contexts).

**Comment by hftf** [comment]
Thank you all for the feedback! I revised the pull request to take into account three changes (`archaic` → `very formal`, Google Books Ngrams URL → Cambridge dictionary source, removed the incomplete complex `bring about` suggestion).

**Pull request merged**